src/query/procedure/mg_procedure_impl.cpp

```diff
@@ -1910,24 +1942,32 @@ mgp_error mgp_edges_iterator_next(mgp_edges_iterator *it, mgp_edge **result) {
   return WrapExceptions(
       [it] {
         MG_ASSERT(it->in || it->out);
-        auto next = [&](auto *impl_it, const auto &end) -> mgp_edge * {
+        auto next = [it](const bool for_in) -> mgp_edge * {
+          auto impl_it = for_in ? it->in_it : it->out_it;
+          const auto end = for_in ? it->in->end() : it->out->end();
```

Comment:
I like what you did here, really nice!

Reply:
I took me some time to wrap my head around it, but it seems much much nicer to pass a single boolean as a parameter, rather that working with those sausage-like decltype/declvalues mumbo-jumbo. The iterators are of the same type, and it turned out i only needed to decide which one to iterate on... So, iterate on first? No? Ok, on second then. BTW, You missed a subtle bug, that originates in line 1946 :) The bug was actually the biggest drawback of using this approach, since it's not so obvious at first. When i created impl_it, i actually created a copy of the underlying iterator, so when i iterated on impl_it, i actually iterated on the copy. The member iterator remained stationary. I've fixed this in subsequent PRs, with making `auto impl_it` -> `auto &impl_it`